linkerd/proxy/transport/src/addrs.rs

```diff
@@ -155,3 +158,23 @@ impl<T: fmt::Display> fmt::Display for Remote<T> {
         self.0.fmt(f)
     }
 }
+
+// === impl TargetPort ===
+
+impl From<OrigDstAddr> for TargetPort {
+    fn from(OrigDstAddr(addr): OrigDstAddr) -> Self {
+        Self(addr.port())
+    }
+}
+
+impl From<SocketAddr> for TargetPort {
+    fn from(addr: SocketAddr) -> Self {
+        Self(addr.port())
+    }
+}
+
+impl linkerd_metrics::FmtLabels for TargetPort {
+    fn fmt_labels(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "target_port=\"{}\"", self.0)
+    }
+}
```

Comment:
I don't think we should have the target types tightly bound to the metric labels like this... I.e. we don't want to expect that the same target types will always be rendered as the same labels, I think.